### PR TITLE
Run Ansible Lint action only on PR and push to main

### PR DESCRIPTION
**Pull request by bluikko**
Since the action is slow at 4 minutes, don't do it on every push and frustrate everyone. Discussion in #24.

On the other hand, it could lead to PRs that will get additional commits if the change was not merged by the contributor first.

Also increase the timeout from 7 minutes to 8. That should be enough (current runs seem to max out at just over 4 minutes).

### CODE REVIEW

**Comment by NeilHanlon** [comment]
@bluikko so we're spending a lot of time building this container to lint every time? That is dumb.

Is there movement on the action issue to make the image pre built? Can you tldr it if so?

**Comment by bluikko** [comment]
> @bluikko so we're spending a lot of time building this container to lint every time? That is dumb.
> 
> Is there movement on the action issue to make the image pre built? Can you tldr it if so?
> 
The issue: https://github.com/ansible/ansible-lint-action/issues/36

tl;dr: If the official Ansible Docker image at hub.docker.com would include ansible-lint then the build should be much simpler.

Discussion there seems to indicate that it should be the "ansible" user at Docker hub that should include the ansible-lint bits in the Docker image. I have not looked into this personally.

ansible-lint is included in the `ansible/molecule` image but that is over 300 MB, and discussion from #ansible-devel referred in the issue says that the image was out of date anyways (1 month ago) due to pipeline errors.

Another option would probably be using a non-official image such as https://hub.docker.com/r/yokogawa/ansible-lint

> Edit: I don't want to get rid of checks on push because it takes too long. It's bad practice and I don't want to get started with that here. Let's fix upstream if we need to.
> 
That would be the best obviously.

> Thanks for the work you're doing!
Likewise!!

**Comment by NeilHanlon** [comment]
@bluikko we have a Rocky Linux docker hub acct. I forked the action repo and am going to setup an auto build and ask you to test using it from this action.

**Comment by dpavlos** [comment]
Be aware of the [pull rate limits](https://www.docker.com/blog/what-you-need-to-know-about-upcoming-docker-hub-rate-limiting/) recently introduced on docker hub. Why not pre-built the image and host it in a local docker registry?

**Comment by hbjydev** [comment]
rocky-linux/ansible-lint-action is a fork that now uses a fixed container image, so this problem can go away as long as we use that instead of the upstream one.

**Comment by NeilHanlon** [comment]
Fixed in #578

**Comment by bluikko** [comment]
> Be aware of the [pull rate limits](https://www.docker.com/blog/what-you-need-to-know-about-upcoming-docker-hub-rate-limiting/) recently introduced on docker hub. Why not pre-built the image and host it in a local docker registry?

This is not an issue, GitHub caches the images.
